src/conversations/Conversations.ts

```diff
@@ -43,7 +44,10 @@ export class ConversationCache {
   async load(loader: CacheLoader) {
     const release = await this.mutex.acquire()
     try {
-      const newConvos = await loader({ latestSeen: this.latestSeen })
+      const newConvos = await loader({
+        latestSeen: this.latestSeen,
+        existing: this.conversations,
```

Comment:
I now provide the existing items in the cache to the loader function, so that `newConversation` can avoid doing unnecessary network requests if the existing conversation is already cached.